### PR TITLE
feat: Configure KameletBinding & Pipe properties

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #149

### CODE REVIEW

File: packages/ui/src/models/visualization/flows/kamelet-schema.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    const definition = KameletSchemaService.getKameletDefinition(stepModel);
+    return {
+      title: definition?.metadata.name || '',
+      schema: KameletSchemaService.getSchemaFromKameletDefinition(definition),
+      definition: stepModel?.ref?.properties || {},
+    };
+  }
+
+  static getSchemaFromKameletDefinition(definition: IKameletDefinition | undefined): JSONSchemaType<unknown> {
```
Comment (lordrip) [inline packages/ui/src/models/visualization/flows/kamelet-schema.service.ts:21]: [very-minor] Could we please make these methods private if they are not intended to be used outside of this class?

**Review by lordrip — APPROVED** [review]
LGTM, just a very minor details around the code

File: packages/ui/src/models/visualization/flows/kamelet-binding.ts [hunk 2]
```diff
@@ -46,15 +49,27 @@ export class KameletBinding implements BaseVisualCamelEntity {
...
+    const rootNode = this.getVizNodeFromStep(this.route.spec?.source, 'source');
+    const stepNodes = this.route.spec?.steps ? this.getVizNodesFromSteps(this.route.spec?.steps) : undefined;
+    const sinkNode = this.getVizNodeFromStep(this.route.spec?.sink, 'sink');
 
-    if (vizNodes !== undefined) {
-      const firstVizNode = vizNodes[0];
-      if (firstVizNode !== undefined) {
-        rootNode.setNextNode(firstVizNode);
-        firstVizNode.setPreviousNode(rootNode);
+    if (stepNodes !== undefined) {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/visualization/flows/kamelet-binding.ts:56]: WebStorm complains with that, saying the argument of `getVizNodesFromSteps()` is incompatible if it's `undefined`, so I think in that case the argument needs to be marked as optional with "?"

File: packages/ui/src/models/visualization/flows/kamelet-binding.ts [hunk 3]
```diff
@@ -21,16 +19,21 @@ export class KameletBinding implements BaseVisualCamelEntity {
     return '';
   }
 
-  getComponentSchema(): VisualComponentSchema | undefined {
-    return undefined;
+  getComponentSchema(path?: string): VisualComponentSchema | undefined {
+    if (!path) return undefined;
+    const stepModel = get(this.route.spec, path) as KameletBindingStep;
+    return stepModel ? KameletSchemaService.getVisualComponentSchema(stepModel) : undefined;
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/visualization/flows/kamelet-binding.ts]: done!

File: packages/ui/src/models/visualization/flows/kamelet-binding.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/visualization/flows/kamelet-binding.ts:56]: added the array check anyway

**Pull request merged**